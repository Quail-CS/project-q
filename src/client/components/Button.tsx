/**
 * @module Button.tsx
 * @description Clickable Button
 */

import * as React from 'react';

interface ButtonProps {
  className: string,
  children: string,
  onClick: any, // TODO: Specify
}

const Button = (props: ButtonProps) => {
  const { className, onClick, children } = props
  return (
    <div className={className} onClick={onClick}>
      {children}
    </div>
  )
}

export default Button;
