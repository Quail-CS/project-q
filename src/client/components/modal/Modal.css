.modal-outer {
  position: absolute;
  height: 100%;
  left: 0;
  top: 0;
  width: 100%;

  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-opacity {
  position: absolute;
  height: 100%;
  left: 0;
  top: 0;
  width: 100%;

  background: #000000;
  opacity: 0.75;

  z-index: 1;
}

.modal-inner {
  background: #212121;
  border: 1px solid #888888;
  padding: 1vh 4vw;
  width: 40vw;

  z-index: 2;
}

.modal-button-container {
  display: flex;
  justify-content: space-between;
  padding-bottom: 4vh
}

.modal-button {
  color: #888888;
  display: inline-block;
  font-size: 1.5rem;
  font-weight: bold;
  padding: 0 5vw;
}

.modal-button:hover {
  color: #FFFFFF;
  cursor: pointer;
}

.modal-body-text {
  color: #A5A8AA;
  display: block;
  font-family: 'Lato', serif;
  padding-bottom: 4vh;
}

.modal-header-text {
  color: #A5A8AA;
  display: block;
  font-size: 2rem;
  font-weight: bold;
  padding: 4vh 0;
}

.modal-file-input {
  width: 0.1px;
  height: 0.1px;
  opacity: 0;
  overflow: hidden;
  position: absolute;
  z-index: -1;
}