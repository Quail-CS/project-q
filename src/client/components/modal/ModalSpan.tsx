/**
 * @module ModalSpan
 * @description Modal Span React Component
 */

import * as React from 'react';

interface IModalSpanProps {
  children: string;
  header: boolean
};

const ModalSpan = (props: IModalSpanProps): JSX.Element => {
  const { children, header } = props;
  const spanClass = header ? 'modal-header-text' : 'modal-body-text';
  return (
    <span className={spanClass}>
      {children}
    </span>
  );
};

export default ModalSpan;
