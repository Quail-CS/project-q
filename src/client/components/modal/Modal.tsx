/**
 * @module Modal
 * @description Modal React Container
 */

import * as React from 'react';

import { Link } from 'react-router-dom';

import modalText from './modalText';
import Button from "../Button";
import ModalSpan from './ModalSpan';

import './Modal.css';

interface IModalProps {
  clearModal: () => any;
  modalType: string;
}

const Modal = (props: IModalProps) => {
  const {
    clearModal,
    modalType,
  } = props;
  let bodyText: string;
  let headerText: string;
  let buttonText: string;

  switch (modalType) {
    case 'reset':
      headerText = modalText.RESET_MODAL_HEADER_TEXT;
      bodyText = modalText.RESET_MODAL_BODY_TEXT;
      buttonText = modalText.RESET_BUTTON_TEXT;
      break;
    default:
      bodyText = '';
      headerText = '';
      buttonText = '';
  }

  return (
    <React.Fragment>
      <div className="modal-opacity" onClick={clearModal}></div>
      <div className="modal-outer">
        <div className="modal-inner">
          <ModalSpan header={true} >{headerText}</ ModalSpan>
          <ModalSpan header={false}>{bodyText}</ ModalSpan>
          <div className="modal-button-container">
            <Button className="modal-button" onClick={clearModal}>CANCEL</ Button>
            <Link to='/account/restart'><Button className="modal-button" onClick={clearModal}>{buttonText}</ Button></Link>
          </div>
        </div>
      </div>
    </React.Fragment>
  )
}

export default Modal;
