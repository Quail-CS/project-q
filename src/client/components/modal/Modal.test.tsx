/**
 * @module Modal.test
 * @description Unit Tests for Modal React Container
 */

import * as Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from 'enzyme';
import * as React from 'react';
import * as Renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router';
import Modal from './Modal';
import ModalSpan from './ModalSpan';
import Button from '../Button';

Enzyme.configure({ adapter: new Adapter() });

// Declare constants and mocks for testing
const mockFn = jest.fn();

describe('Reset Modal Component', () => {
  it('matches the snapshot', () => {
    const tree = Renderer.create(
      <MemoryRouter>
        <Modal clearModal={mockFn} modalType='reset' />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders two divs - one with class "modal-opacity" and one with class "modal-outer"', () => {
    const shallowModalWrapper = Enzyme.shallow(<Modal clearModal={mockFn} modalType='reset' />);
    expect(shallowModalWrapper.find('.modal-opacity').type()).toEqual('div');
    expect(shallowModalWrapper.find('.modal-outer').type()).toEqual('div');
    expect(shallowModalWrapper.find('.modal-inner').type()).toEqual('div');
  });

  it('renders two <ModalSpan>s and two <Button>s inside modal-inner"', () => {
    const shallowModalWrapper = Enzyme.shallow(<Modal clearModal={mockFn} modalType='reset' />);
    expect(shallowModalWrapper.find('.modal-inner').find(ModalSpan)).toHaveLength(2);
    expect(shallowModalWrapper.find('.modal-inner').find(Button)).toHaveLength(2);
  });
});
