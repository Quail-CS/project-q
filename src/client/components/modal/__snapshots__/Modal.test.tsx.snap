// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Modal Component matches the snapshot 1`] = `
Array [
  <div
    className="modal-opacity"
    onClick={[MockFunction]}
  />,
  <div
    className="modal-outer"
  >
    <div
      className="modal-inner"
    >
      <span
        className="modal-header-text"
      >
        Reset Ethiq?
      </span>
      <span
        className="modal-body-text"
      >
        Clicking "Continue" will reset your profile and issue selections. Continue?
      </span>
      <div
        className="modal-button-container"
      >
        <div
          className="modal-button"
          onClick={[MockFunction]}
        >
          CANCEL
        </div>
        <a
          href="/account/restart"
          onClick={[Function]}
        >
          <div
            className="modal-button"
            onClick={[MockFunction]}
          >
            CONTINUE
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;
